### PR TITLE
cses-2428 C++ solution added

### PR DESCRIPTION
**Pull request by gsvamsi18**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/cses-2428.mdx [hunk 1]
```diff
@@ -18,7 +18,53 @@ it will create ($i - j$) new subarrays, i.e ($j$ to $i$), ($j + 1$ to $i$), etc.
...
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+int main() {
+	int n, k;
+	cin >> n >> k;
+	vector<int> arr(n);
+	for (int i = 0; i < n; i++) 
+		cin >> arr[i];
```
Comment (SansPapyrus683) [inline solutions/cses-2428.mdx:31]: braces

File: solutions/cses-2428.mdx [hunk 2]
```diff
@@ -18,7 +18,53 @@ it will create ($i - j$) new subarrays, i.e ($j$ to $i$), ($j + 1$ to $i$), etc.
 **Time Complexity:** $\mathcal{O}(N)$
 
 <LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+int main() {
+	int n, k;
```
Comment (SansPapyrus683) [inline solutions/cses-2428.mdx:27]: declarations on separate lines

File: solutions/cses-2428.mdx [hunk 3]
```diff
@@ -18,7 +18,53 @@ it will create ($i - j$) new subarrays, i.e ($j$ to $i$), ($j + 1$ to $i$), etc.
...
+	cin >> n >> k;
+	vector<int> arr(n);
+	for (int i = 0; i < n; i++) 
+		cin >> arr[i];
+	int left = 0, right = 0;
```
Comment (SansPapyrus683) [inline solutions/cses-2428.mdx:32]: same here

File: solutions/cses-2428.mdx [hunk 4]
```diff
@@ -18,7 +18,53 @@ it will create ($i - j$) new subarrays, i.e ($j$ to $i$), ($j + 1$ to $i$), etc.
...
+	long long ans = 0;
+	int distinct = 0;
+	map<int, int> freq;
+	while (left < n) {
+		while (right < n) {
+			if (freq.count(arr[right]) == 0  && distinct == k)
+				break;
```
Comment (SansPapyrus683) [inline solutions/cses-2428.mdx:39]: braces here

File: solutions/cses-2428.mdx [hunk 5]
```diff
@@ -18,7 +18,53 @@ it will create ($i - j$) new subarrays, i.e ($j$ to $i$), ($j + 1$ to $i$), etc.
...
+		// Slide the window rightwards.
+		if (freq[arr[left]] == 1) {
+			distinct--;
+			freq.erase(arr[left]);
+		} else {
+			freq[arr[left]] -= 1;
+		}
+		left++;
+	}
+	cout << ans;
```
Comment (SansPapyrus683) [inline solutions/cses-2428.mdx:63]:
```suggestion
	cout << ans << endl;
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Comment by gsvamsi18** [comment]
can I ask why we need to declare the variables on separate lines, is it for better readability?

**Comment by SansPapyrus683** [comment]
> can I ask why we need to declare the variables on separate lines, is it for better readability?

yeah

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/cses-2428.mdx [hunk 6]
```diff
@@ -18,7 +18,60 @@ it will create ($i - j$) new subarrays, i.e ($j$ to $i$), ($j + 1$ to $i$), etc.
 **Time Complexity:** $\mathcal{O}(N)$
 
 <LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
```
Comment (Aadit-Ambadkar) [inline solutions/cses-2428.mdx:26]:
```suggestion
typedef long long ll

```

File: solutions/cses-2428.mdx [hunk 7]
```diff
@@ -18,7 +18,60 @@ it will create ($i - j$) new subarrays, i.e ($j$ to $i$), ($j + 1$ to $i$), etc.
...
+	int k;
+	cin >> n >> k;
+	vector<int> arr(n);
+	for (int i = 0; i < n; i++) { 
+		cin >> arr[i];
+	}
+
+	int left = 0;
+	int right = 0;
+	long long ans = 0;
```
Comment (Aadit-Ambadkar) [inline solutions/cses-2428.mdx:38]:
```suggestion
	ll ans = 0;
```

**Review by Aadit-Ambadkar — APPROVED** [review]

File: solutions/cses-2428.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/cses-2428.mdx:26]: (optional)

**Pull request merged**